homeassistant/components/climate/demo.py

```diff
@@ -108,9 +123,14 @@ def fan_list(self):
         """List of available fan modes."""
         return self._fan_list
 
-    def set_temperature(self, temperature):
-        """Set new target temperature."""
-        self._target_temperature = temperature
+    def set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        if self._target_temperature:
+            self._target_temperature = kwargs.get(ATTR_TEMPERATURE)
+        if self._target_temperature_high:
+            self._target_temperature_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        if self._target_temperature_low:
```

Comment:
Remove the if statements, and just set the temperatures directly. If one is None and no data attribute for that one is passed, it will still be None. The upside is that you can pass a data attribute and change the temp from None to a temp.